coral-trino/src/main/java/com/linkedin/coral/trino/trino2rel/TrinoSqlToRelConverter.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2017-2022 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.trino2rel;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.calcite.linq4j.Ord;
+import org.apache.calcite.plan.Convention;
+import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.RelOptTable;
+import org.apache.calcite.plan.RelOptUtil;
+import org.apache.calcite.prepare.Prepare;
+import org.apache.calcite.rel.RelCollation;
+import org.apache.calcite.rel.RelCollations;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.RelRoot;
+import org.apache.calcite.rel.core.Uncollect;
+import org.apache.calcite.rel.logical.LogicalValues;
+import org.apache.calcite.rel.metadata.JaninoRelMetadataProvider;
+import org.apache.calcite.rel.metadata.RelMetadataQuery;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlExplainFormat;
+import org.apache.calcite.sql.SqlExplainLevel;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlUnnestOperator;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.sql2rel.SqlRexConvertletTable;
+import org.apache.calcite.sql2rel.SqlToRelConverter;
+
+import com.linkedin.coral.hive.hive2rel.functions.HiveExplodeOperator;
+import com.linkedin.coral.hive.hive2rel.rel.HiveUncollect;
+
+
+/**
+ * Class to convert Trino SQL to Calcite RelNode. This class
+ * specializes the functionality provided by {@link SqlToRelConverter}.
+ */
+class TrinoSqlToRelConverter extends SqlToRelConverter {
+
+  TrinoSqlToRelConverter(RelOptTable.ViewExpander viewExpander, SqlValidator validator,
+      Prepare.CatalogReader catalogReader, RelOptCluster cluster, SqlRexConvertletTable convertletTable,
+      Config config) {
+    super(viewExpander, validator, catalogReader, cluster, convertletTable, config);
+  }
+
+  // This differs from base class in two ways:
+  // 1. This does not validate the type of converted rel rowType with that of validated node. This is because
+  //    hive is lax in enforcing view schemas.
+  // 2. This skips calling some methods because (1) those are private, and (2) not required for our usecase
```

Comment:
It is not clear what that means.

Reply:
Is this copied from `HiveSqlToRelConverter`?

Reply:
Yes it's copied from HiveSqlToRelConverter.